daemon/containerd/image_delete.go

```diff
@@ -35,6 +41,24 @@ import "github.com/docker/docker/api/types"
 // If prune is true, ancestor images will each attempt to be deleted quietly,
 // meaning any delete conflicts will cause the image to not be deleted and the
 // conflict will not be reported.
-func (i *ImageService) ImageDelete(imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error) {
-	panic("not implemented")
+func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error) {
+	// TODO(thaJeztah): implement ImageDelete "force" options; see https://github.com/moby/moby/issues/43850
+	// TODO(thaJeztah): implement ImageDelete "prune" options; see https://github.com/moby/moby/issues/43849
```

Comment:
Per discussion in the maintainers meeting; we need to add more tracking issues for; partial IDs etc. (some of that logic is implemented in `GetImage()` as part of https://github.com/moby/moby/pull/43818)